rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -555,20 +557,58 @@ def decode_transaction_hashes(
                 tx_hashes = [EVMTxHash(x[0]) for x in cursor]
 
         for tx_hash in tx_hashes:
-            try:
-                receipt = self.transactions.get_or_query_transaction_receipt(tx_hash)
-            except RemoteError as e:
-                raise InputError(f'{self.evm_inquirer.chain_name} hash {tx_hash.hex()} does not correspond to a transaction') from e  # noqa: E501
-
-            # TODO: Change this if transaction filter query can accept multiple hashes
-            with self.database.conn.read_ctx() as cursor:
-                txs = self.dbevmtx.get_evm_transactions(
-                    cursor=cursor,
-                    filter_=EvmTransactionsFilterQuery.make(tx_hash=tx_hash, chain_id=self.evm_inquirer.chain_id),  # noqa: E501
-                    has_premium=True,  # ignore limiting here
+            if tx_hash == GENESIS_HASH:
```

Comment:
wtf? From line 560 to line 596 you just copied paste the genesis tx hash code from `get_or_query_transaction_receipt()`. In what universe do you think this is acceptable?

 This needs to be abstracted into its own function and called in both places properly.

Code that does such sloppy copy pasta would never be accepted into the repo. Don't ask for review with code like this.